diesel/src/types/mod.rs

```diff
@@ -337,18 +338,100 @@ pub enum IsNull {
     No,
 }
 
+#[derive(Debug, Clone, Copy)]
+#[doc(hidden)]
+pub struct ToSqlOutput<'a, T, DB> where
```

Comment:
> I don't see any advantage over passing those fields directly as arguments to to_sql.

Well there's two main points (I explained some of this in the PR description, not sure if you read it). The big one is that less existing code will need to change this way. If we add a second parameter, we'll need to modify every definition *and* every use of it. The second reason I'm preferring this is that I don't want to have to modify all this code *again* when we realize we needed a third or a fourth parameter.

> Creating this wrapping struct needs ~100 lines boilerplate code

Lines of code is not a valuable metric to measure things against.